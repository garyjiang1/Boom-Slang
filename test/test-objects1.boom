class MyClass:
	#adding some comments
	required:
		int one
		# adding some comments
		string two
		MyClass recursive1

	optional: /# more comments #/
		float three = 3.0
		float four = 4.0
		float five  = 5.0 # adding some comments
		char six = '6'
		MyClass recursive2 = MyClass() # adding some comments

	def construct(): /# here
are some more comments designed to be annoying #/
		println("called custom constructor")

	def to_string() returns string: /#
 another one #/
		# adding some comments
		return "this is my custom tostring"
# adding some comments
	def method1(int a, int b, int z) returns int:
		println("called method1") /#

		more    stupid
		comments
		#/
		int q = a + b + z
		int z = q
		return z

	# adding some comments
	def method2():
    # adding some comments
		println("called method2")

println("printing some random stuff")
MyClass foo = MyClass(1, "two", MyClass())
println(foo)
println(foo.one)
println(foo.two)
println(foo.three)
println(foo.four)
println(foo.five)
println(foo.six)
MyClass recursive1 = foo.recursive1
MyClass recursive2 = foo.recursive2
println(recursive1)
println(recursive2.to_string())
println(foo.method1(1, 2, 3))
foo.method2()
