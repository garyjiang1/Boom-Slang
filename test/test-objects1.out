printing some random stuff
called custom constructor
called custom constructor
this is my custom tostring
1
two
3.0000
4.0000
5.0000
6
this is my custom tostring
this is my custom tostring
called method1
6
called method2
